### DISCUSSION TITLE
accessing objects as graphql source from within a DataFetcher

### DISCUSSION THREAD

**Discussion by softprops** [graphql-java/graphql-java#2581] [Q&A]
Is there a way to serialize a Document / SelectionSet back into its source string in GraphQL IDL format? 

I have a resolver that I'd like to send an external server to replicate the request. 

I could do this using a custom context object and have my server pass along the raw request body query but it seems like I already have access to most of the same data already in the provided data fetching environment. I'm just not sure to to serialize it back into its native format

**Comment by softprops** [comment]
if anyone else bumps into this `AstPrinter` is what you want

```java
DataFetcher<String> reflect() {
    return env -> graphql.language.AstPrinter.printAstCompact(env.getDocument());
}
````

**Comment by knotenpunkt** [comment]
good question @softprops, in that i am also interessted in :+1: .
but for me it would be also a good extension not only forward the request to a replicate, but also to change parameter and selectionset-data (sometimes)

in the case i only forward i do the thing what you suggested, put it into the context
in the case i must manipulate the data, yeah for that caes i build a new graphql-query from the data i get from the datafetchingenvironment. The problem here, i can only reuse a small subset of expression-power of graphql.

So i am also interessted if there is a better solution out there

**Comment by softprops** [comment]
That's actually my use case as well. Ill let you know what I find but I'll keep an eye on this discussion to see if others have ideas as well
